tagstudio/src/core/library/alchemy/visitors.py

```diff
@@ -38,7 +38,18 @@ def visit_constraint(self, node: Constraint) -> ColumnExpressionArgument:
         elif node.type == ConstraintType.FileType:
             return Entry.suffix.ilike(node.value)
 
+        # raise exception if Constraint stays unhandled
         raise NotImplementedError("This type of constraint is not implemented yet")
 
     def visit_property(self, node: Property) -> None:
         return
+
+    def visit_not(self, node: Not) -> ColumnExpressionArgument:
+        return ~Entry.id.in_(
+            # TODO TSQLANG there is technically code duplication from Library.search_library here
```

Comment:
Not really sure on whether / how I should deal with this duplication, opinions would be appreciated